apps/snitch_core/test/data/model/tax_rate_test.exs

```diff
@@ -0,0 +1,147 @@
+    value: 0.5,
+    included_in_price: false,
+    calculator: "DefaultCalculator"
+  }
+
+  describe "create/1" do
+    test "tax rate successfully" do
+      params = create_tax_params()
+      assert {:ok, _} = TaxRateModel.create(params)
+    end
+
+    test "fails, name missing" do
+      params = create_tax_params()
+      params = Map.drop(params, [:name])
+      assert {:error, cs} = TaxRateModel.create(params)
+      assert %{name: ["can't be blank"]} = errors_on(cs)
+    end
+
+    test "fails, calculator missing" do
+      params = create_tax_params()
+      params = Map.drop(params, [:calculator])
+      assert {:error, cs} = TaxRateModel.create(params)
+      assert %{calculator: ["can't be blank"]} = errors_on(cs)
+    end
+  end
+
+  describe "update/2" do
+    setup :tax_rate
+
+    test "successfuly calculator field", context do
+      params = %{calculator: "FlatRateCalculator"}
+      %{tax_rate: tr} = context
+      assert {:ok, tr_new} = TaxRateModel.update(params, tr)
+      assert tr_new.id == tr.id
+      refute tr_new.calculator == tr.calculator
+    end
+
+    test "successful without instance", context do
+      %{tax_rate: tr} = context
+      params = %{name: "Germany", id: tr.id}
+      assert {:ok, tr_new} = TaxRateModel.update(params)
+      assert tr_new.id == tr.id
+      refute tr_new.name == tr.name
+    end
+
+    test "failed, name missing", context do
+      %{tax_rate: tr} = context
+      params = %{name: "", id: tr.id}
+      assert {:error, cs} = TaxRateModel.update(params)
+      assert %{name: ["can't be blank"]} = errors_on(cs)
+    end
+  end
+
+  describe "delete/1" do
+    setup :tax_rate
+
+    test "tax category successfully, with instance", context do
+      %{tax_rate: tr} = context
+      assert {:ok, tr_deleted} = TaxRateModel.delete(tr)
+      refute is_nil(tr_deleted.deleted_at)
+      tr_received = Repo.get(TaxRate, tr_deleted.id)
+      refute is_nil(tr_received)
+    end
+
+    test "tax category successfully, with id", context do
+      %{tax_rate: tr} = context
+      assert {:ok, tr_deleted} = TaxRateModel.delete(tr.id)
+      refute is_nil(tr_deleted.deleted_at)
+      tr_received = Repo.get(TaxRate, tr_deleted.id)
+      refute is_nil(tr_received)
+    end
+  end
+
+  describe "get/2" do
+    setup :tax_rate
+
+    test "tax rate", context do
+      %{tax_rate: tr} = context
+      tr_ret = TaxRateModel.get(tr.id)
+      assert tr.id == tr_ret.id
+    end
+
+    test "no tax rate, is deleted", context do
+      %{tax_rate: tr} = context
+      assert {:ok, tr} = TaxRateModel.delete(tr)
+      refute is_nil(tr.deleted_at)
+      tr_ret = TaxRateModel.get(tr.id)
+      assert is_nil(tr_ret)
+    end
+
+    test "tax category, is deleted", context do
+      %{tax_rate: tr} = context
+      {:ok, tr} = TaxRateModel.delete(tr)
+      refute is_nil(tr.deleted_at)
+      tr_ret = TaxRateModel.get(tr.id, false)
+      refute is_nil(tr_ret)
+    end
+  end
+
+  describe "get_all/1" do
```

Comment:
All these asserts can be made in just one test case, where one of the tax rates has been soft deleted.